plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py

```diff
@@ -1,177 +1,169 @@
+        # =>  cybox_types = ['domain-name']
+        il = pattern.inspect()
+        cybox_types = list(il.comparisons.keys())
+        return (
+            len(il.observation_ops) == 0
+            and len(il.qualifiers) == 0
+            and len(il.comparisons) == 1
+            and len(cybox_types) == 1  # must be point-indicator (one field only)
+            and len(il.comparisons[cybox_types[0]][0])
+            == 3  # ('value', '=', 'evil.com')
+            and il.comparisons[cybox_types[0]][0][1] == "="  # equality comparison
+        )
+    except Exception:
+        return False
 
 
-def map_to_broker(msg, module_namespace):
-    """Maps the internal message format to a broker message.
-    @param msg The message that shall be converted
-    @param module_namespace A Zeek namespace to use for event sending
+def map_indicator_to_broker_event(
+    indicator: Indicator, module_namespace: str, logger
+) -> Union[broker.zeek.Event, None]:
+    """
+    Maps STIX-2 Indicators to Broker events using the Zeek Intel format
+    @see https://docs.zeek.org/en/current/scripts/base/frameworks/intel/main.zeek.html#type-Intel::Type
+    @param indicator The STIX-2 Indicator to convert
+    @param module_namespace A Zeek namespace to use for sending the event
     @return The mapped broker event or None
     """
-    msg_type = type(msg).__name__.lower()
-    if msg_type == "sighting":
-        # convert sighting to zeek event
-        return broker.zeek.Event(
-            f"{module_namespace}::sighting",
-            (msg.ts, str(msg.intel), msg.context),
+    if type(indicator) is not Indicator:
+        if logger:
+            logger.debug(f"Discarding message, expected STIX-2 Indicator: {indicator}")
+        return None
+
+    if not is_point_equality_ioc(indicator.pattern):
+        logger.debug(
+            f"Zeek only supports point-IoCs. Cannot map compound pattern to a Zeek Intel item: {indicator.pattern}"
         )
-    elif msg_type == "intel":
-        # convert intel to zeek event
-        intel_data = map_intel_data_to_broker(msg.data)
-        if not intel_data:
-            return None
-        return broker.zeek.Event(
-            f"{module_namespace}::intel",
-            (msg.ts, str(msg.id), intel_data, msg.operation.value),
+        return None
+
+    # pattern is in the form [file:name = 'foo']
+    (object_path, ioc_value) = indicator.pattern[1:-1].split("=")
+    object_path, ioc_value = object_path.strip(), ioc_value.strip()
+    if ioc_value.startswith("'") and ioc_value.endswith("'"):
+        ioc_value = ioc_value[1:-1]
+
+    # get matching Zeek intel type
+    zeek_type = zeek_intel_type_map.get(object_path, None)
+    if not zeek_type:
+        logger.debug(
+            f"No matching Zeek type found for STIX-2 indicator type '{object_path}'"
         )
+        return None
+
+    if zeek_type == "URL":
+        # remove leading protocol, if any
+        ioc_value = re.sub(r"^https?://", "", ioc_value)
```

Comment:
Couldn't this be any protocol, like `ftp://`?

Reply:
yes, you are right, we should remove all protocols.

Reply:
Added a snip to remove all URL schemes.